src/Symfony/Component/ObjectMapper/CollectionMapper.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper;
+
+use Symfony\Component\ObjectMapper\Exception\MappingException;
+use Symfony\Component\ObjectMapper\Exception\WrappedMappingException;
+
+/**
+ * Map a collection of objects using the ObjectMapper.
+ *
+ * @experimental
+ *
+ * @author Martin Komischke <martin.komischke@gmail.com>
+ */
+final class CollectionMapper implements CollectionMapperInterface
+{
+    public function __construct(
+        private ObjectMapperInterface $mapper,
+        private string $throwPolicy = CollectionMapperThrowPolicy::FAIL_SAFE,
+    ) {
+    }
+
+    public function map(iterable $sourceCollection, array|string|null $target = null): \Generator
+    {
+        if (\is_array($target)) {
+            yield from $this->mapArray($sourceCollection, $target);
+        } else {
+            return match ($this->throwPolicy) {
```

Comment:
`return` is not necessary as the values are returned by `yield`.
```suggestion
           match ($this->throwPolicy) {
```